app/controllers/stories_controller.rb

```diff
@@ -330,4 +331,58 @@ def assign_podcasts
   def assign_classified_listings
     @classified_listings = ClassifiedListing.where(published: true).select(:title, :category, :slug, :bumped_at)
   end
+
+  def set_json_ld
+    @user_json_ld = {
+      "@context": "http://schema.org",
+      "@type": "Person",
+      "mainEntityOfPage": {
+        "@type": "WebPage",
+        "@id": "<%= user_url(@user) %>"
+      },
+      "url": "<%= user_url(@user) %>",
+      "sameAs": [],
+      "image": "<%= ProfileImage.new(@user).get(width: 320) %>",
+      "name": "<%= @user.name %>",
+      "email": "",
+      "jobTitle": "",
+      "description": "<%= @user.summary.presence || ['404 bio not found'].sample %>",
+      "disambiguatingDescription": [],
+      "worksFor": [
+        {
+          "@type": "Organization"
+        },
+      ],
+      "alumniOf": ""
+    }
+    set_user_profile_json_ld
+    set_user_same_as_json_ld
+  end
+
+  def set_user_profile_json_ld
+    @user_json_ld[:disambiguatingDescription].append(@user.mostly_work_with) if @user.mostly_work_with.present?
+    @user_json_ld[:disambiguatingDescription].append(@user.currently_hacking_on) if @user.currently_hacking_on.present?
+    @user_json_ld[:disambiguatingDescription].append(@user.currently_learning) if @user.currently_learning.present?
+    @user_json_ld[:worksFor][0][:employer_name] = @user.employer_name if @user.employer_name.present?
+    @user_json_ld[:worksFor][0][:employer_url] = @user.employer_url if @user.employer_url.present?
+    @user_json_ld[:alumniOf].concat(@user.education) if @user.education.present?
+    @user_json_ld[:email].concat(@user.email) if @user.email_public
+    @user_json_ld[:jobTitle].concat(@user.employment_title) if @user.employment_title.present?
```

Comment:
these are just string keys, you can use, for example: 

```ruby
@user_json_ld[:alumniOf] = @user.education if @user.education.present?
```

no need for concat